### PR TITLE
Add ability to set components after initialized. 

### PR DESCRIPTION
**Pull request by noblica**
Currently it is not possible to set or access the components map, after `storyblokInit` has been executed.

This PR adds and exposes the `setComponents` method, so that components can be set at some point after initialization.
~~It also exposes the `componentsMap` variable, so it can be appended if needed~~

I've also taken the liberty of updating the next-demo to reflect the new usage, and the README.md where I thought it made sense. Please let me know if this is fine or if more work is needed here.

Related issue: https://github.com/storyblok/storyblok-react/issues/595

### CODE REVIEW

**Review by arorachakit — APPROVED** [review]

**Review by fgiuliani — APPROVED** [review]

File: lib/common/index.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import {
 } from "../types";
 
 let storyblokApiInstance: StoryblokClient = null;
-let componentsMap: SbReactComponentsMap = {};
+export let componentsMap: SbReactComponentsMap = {};
```
[the remark this answers is not in the document]
Comment (noblica) [reply lib/common/index.ts:10]: Why create an extra function, instead of just exposing it as-is?

[the remark this answers is not in the document]
Comment (alexjoverm) [reply lib/common/index.ts:10]: Just encapsulation and reference-keeping - I see your point on appending, but as components are set externally, the user has full control of the object reference.

btw, thanks to you for the PR, it's cool and to the point!

[the remark this answers is not in the document]
Comment (noblica) [reply lib/common/index.ts]: Ah you are right! I have removed the export from the PR.
Thanks for the kind words :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (fgiuliani) [reply lib/common/index.ts]: Thank you!

**Pull request merged**